polkadot/runtime/westend/src/xcm_config.rs

```diff
@@ -142,14 +145,15 @@ impl Contains<Location> for OnlyParachains {
 	}
 }
 
-pub struct CollectivesOrFellows;
-impl Contains<Location> for CollectivesOrFellows {
+pub struct SystemOrFellows<ParaId>(PhantomData<ParaId>);
+impl<ParaId: IsSystem + From<u32>> Contains<Location> for SystemOrFellows<ParaId> {
 	fn contains(location: &Location) -> bool {
-		matches!(
+		let is_system = IsChildSystemParachain::<ParaId>::contains(location);
```

Comment:
Doesn't this allow any XCM from a system chain to use `UnpaidExecution`?

Reply:
From my understanding no, otherwise the code explicitly allowing the child "fellows" to also not pay would not make much sense (and the could would have been broken before already).

Reply:
https://github.com/paritytech/polkadot-sdk/blob/8428f678fe5179399f6854e194f1c8b30a9102f9/polkadot/xcm/xcm-builder/src/barriers.rs#L320

Reply:
The linked barrier is about "is this a system chain", but here in this context it is about allowing unpaid execution.

Reply:
Yes it does, but only when the origin is a system chain “root”, and not any child sublocations. IMO it is ok to allow it

Reply:
> origin is a system chain “root”

Where is this checked?

Reply:
This means if the origin is any of the system chains or [collectives, fellows] then it can do unpaid execution.
It doesn't allow any account in a system chain to use unpaid execution, if that's what you're worried about.

Reply:
Only "root" can send XCMs with the location of the chain itself, users would appear as sublocations

Reply:
I think now I found out what confused me.

`(0, [Parachain(SYSTEM_ID), AccountId32])`

This would be for example a location if send by a user or?

Reply:
Yes

Reply:
Okay thank you ;) This explains it. I thought we would just allow anything as long as it comes from the system chain. But we actually match explicit, so it doesn't make sense what I was worried about.

Reply:
It was a good question and thought experiment, better safe than sorry 👍🏻

Reply:
> I think now I found out what confused me.
> 
> `(0, [Parachain(SYSTEM_ID), AccountId32])`
> 
> This would be for example a location if send by a user or?

yes, you just need to add `DescendOrigin(AccountId32`) to the XCM program